File: master-node-docker/sentinel/helpers/btc.py
# coding=utf-8
import requests

from ..config import BTC_BASED_COINS


class BTCHelper(object):
    def __init__(self, coins):
        self.coins = coins

    def get_new_address(self, symbol):
        try:
            server = self.coins[symbol]
            url = '{}address'.format(server['url'])
            res = requests.get(url).json()
            return res['address'] if res['success'] else None
        except Exception as error:
            print(error)
            return None

    def get_balance(self, address, symbol):
        try:
            server = self.coins[symbol]
            url = '{}balance?address={}'.format(server['url'], address)
            res = requests.get(url).json()
            return res['balance'] if res['success'] else None
        except Exception as error:
            print(error)
            return None

    def transfer(self, to_address, value, symbol):
        try:
            server = self.coins[symbol]
            url = '{}transfer'.format(server['url'])
            res = requests.post(url, json={
                'toAddress': to_address,
                'value': value
            }).json()
            return res['txHash'] if res['success'] else None
        except Exception as error:
            print(error)
            return None


btc_helper = BTCHelper(BTC_BASED_COINS)

File: master-node-nodejs/src/helpers/btc.js
import axios from "axios";
import { BTC_BASED_COINS } from "../config/swaps";

class BTC_Helper {
  constructor(coins) {
    this.coins = coins;
  }
  getNewAddress(symbol, cb) {
    let server = this.coins[symbol];
    let url = `${server['ip']}/address`;
    axios.get(url)
      .then((resp) => {
        resp = resp.data;
        if (resp['success'])
          cb(resp['address']);
        else
          cb(null);
      })
      .catch((error) => {
        console.log('error');
        cb(null);
      });
  }
  getBalance(address, symbol, cb) {
    let server = this.coins[symbol];
    let url = `${server['ip']}/balance?address=${address}`;
    axios.get(url)
      .then((resp) => {
        resp = resp.data;
        if (resp['success'])
          cb(resp['balance']);
        else
          cb(null);
      })
      .catch((error) => {
        console.log('error');
        cb(null);
      });
  }
  transfer(toAddress, value, symbol, cb) {
    let server = this.coins[symbol];
    let url = `${server['ip']}/transfer`;
    axios.post(url, {
      'toAddress': toAddress,
      'value': value
    }).then((resp) => {
      resp = resp.data;
      if (resp['success'])
        cb(resp['txHash']);
      else
        cb(null);
    }).catch((error) => {
      console.log('error');
      cb(null);
    });
  }
}

export const BTCHelper = new BTC_Helper(BTC_BASED_COINS);